src/pages/Login.js

```diff
@@ -1,32 +1,43 @@
 import styled from "styled-components";
 import { useNavigate } from "react-router-dom";
+import { fromLocalStorage, toLocalStorage } from "../utils/localStorage";
 
-export default function Login() {
+export default function Login({setUser}) {
   const navigate = useNavigate();
 
-  function loginAs(role){
-    localStorage.setItem('user', role);
+  function loginAs(loginName){
+    const userList = fromLocalStorage("users");
+    let user = userList.filter((item) => item.name === loginName);
+    user = user[0];
```

Comment:
You could use the find method here instead of filtering and then assigning the zeroth index to user.

Also, what happens if the user is not found?